src/contexts/providers.context.tsx

```diff
@@ -128,6 +134,51 @@ const ProvidersProvider: FC = (props) => {
     }
   }, [connectedProvider]);
 
+  const changeNetwork = useCallback(
+    async (chainId: string) => {
```

Comment:
I'll change it because in this case is more readable but you are tricking the behaviour, if you don't write `await` is like you forget the detail as write `().then or ().catch`

Reply:
Sure, but with async, you may forget 2 things, the try/catch block, and the await.

Anyway IMO the main reason to avoid using async is not having to add the verbose try/catch block.